protokube/pkg/protokube/aws_volume.go

```diff
@@ -164,73 +164,83 @@ func newEc2Filter(name string, value string) *ec2.Filter {
 func (a *AWSVolumes) findVolumes(request *ec2.DescribeVolumesInput) ([]*Volume, error) {
```

Comment:
So `findVolumes` is used in two places, once when querying all volumes and once when querying by ID.

Although in theory it's fine to filter both by etcd volume, I think it would be cleaner to add the filter at the end here:

```
func (a *AWSVolumes) FindVolumes() ([]*Volume, error) {
	request := &ec2.DescribeVolumesInput{}
	request.Filters = []*ec2.Filter{
		newEc2Filter("tag:"+awsup.TagClusterName, a.clusterTag),
		newEc2Filter("tag-key", awsup.TagNameRolePrefix+awsup.TagRoleMaster),
		newEc2Filter("availability-zone", a.zone),
	}

	return a.findVolumes(request)
}
```

Incidentally, that shows that it is the master role tag that is causing the problems.